packages/apex-parser-ast/src/utils/CaseInsensitiveString.ts

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the
+ * repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * A case-insensitive string wrapper that behaves like a JavaScript string primitive.
+ * Stores the original string and provides case-insensitive comparison methods.
+ */
+export class CaseInsensitiveString {
```

Comment:
I get why we need these data structures for handling case insensitive strings but it feels silly to have to build them from scratch like this. Not that trusting a potentially even stranger third party library that might exist for handling this stuff is better, considering how they might handle odd case sensitive cases differently from what we'd like. I'm rambling though, my point is, maybe we bring some re-usable data structure utilities like this out for use in other packages eventually. Thoughts?